python/cugraph-dgl/tests/nn/test_gatconv.py

```diff
@@ -25,52 +25,126 @@
 torch = import_optional("torch")
 dgl = import_optional("dgl")
 
-options = {
+options1 = {
+    "bipartite": [False, True],
     "idtype_int": [False, True],
     "max_in_degree": [None, 8],
-    "num_heads": [1, 2, 3, 7],
+    "num_heads": [1, 2, 7],
     "to_block": [False, True],
 }
 
 
-@pytest.mark.parametrize(",".join(options.keys()), product(*options.values()))
-def test_gatconv_equality(idtype_int, max_in_degree, num_heads, to_block):
+@pytest.mark.parametrize(",".join(options1.keys()), product(*options1.values()))
```

Comment:
Curious why you did it this way instead of letting pytest produce the cartesian product for you.  That has the advantage of easily adding marks, ids, etc. to individual parameters ([example](https://github.com/rapidsai/cugraph/blob/2c43e3051a65f7047a7e3709410337e83839d119/python/cugraph/cugraph/tests/link_analysis/test_pagerank.py#L161-L168)).  I don't have a strong preference, just wondering if there's an advantage for doing it this way.

Reply:
It is just for convenience (at some cost of readability). I would only need to modify one line of code when adding / removing / renaming a parameter.

Reply:
I changed the code to use pytest cartesian product directly, so that the style is aligned with other cugraph tests.